Assets/MixedRealityToolkit.SDK/Features/UX/Scripts/BoundingBox/BoundingBox.cs

```diff
@@ -130,8 +134,19 @@ public enum RotationHandlePrefabCollider
             Sphere,
             Box
         }
-        #endregion Enums
 
+        /// <summary>
+        /// Container for handle references and states (including scale and rotation type handles)
+        /// </summary>
+        private class Handle
```

Comment:
Why not a struct?

Reply:
Because our handle list is
1) small
2) contains reference types

Making this a struct would require a copy back of the struct every for loop. With List<> you can't do handles[i].state = blah. 

Making the handles member a Handle[] array is annoying because it's not filled in one place. So we need an indexCounter to be tracked/incremented everywhere